tests/robust/test_internals_linearize.py

```diff
@@ -178,3 +174,173 @@ def test_nmc_param_influence_vmap_smoke(
         assert not torch.isnan(v).any(), f"eif for {k} had nans"
         assert not torch.isinf(v).any(), f"eif for {k} had infs"
         assert not torch.isclose(v, torch.zeros_like(v)).all(), f"eif for {k} was zero"
+
+
+@pytest.mark.parametrize(
+    "loc", [torch.zeros(2, requires_grad=True), torch.ones(2, requires_grad=True)]
+)
+@pytest.mark.parametrize(
+    "cov_mat",
+    [
+        torch.eye(2, requires_grad=False),
+        torch.tensor(torch.ones(2, 2) + torch.eye(2), requires_grad=False),
+    ],
+)
+@pytest.mark.parametrize(
+    "v",
+    [
+        torch.tensor([1.0, 0.0], requires_grad=False),
+        torch.tensor([0.0, 1.0], requires_grad=False),
+        torch.tensor([1.0, 1.0], requires_grad=False),
+        torch.tensor([0.0, 0.0], requires_grad=False),
+    ],
+)
+def test_empirical_fisher_vp_against_analytical(
+    loc: torch.Tensor, cov_mat: torch.Tensor, v: torch.Tensor
+):
+    func_log_prob = gaussian_log_prob
+    log_prob_params = {"loc": loc}
+    N_monte_carlo = 10000
+    data = Predictive(GaussianModel(cov_mat), num_samples=N_monte_carlo)(loc)
+    empirical_fisher_vp_func = make_empirical_fisher_vp(
+        func_log_prob, log_prob_params, data, cov_mat=cov_mat
+    )
+
+    empirical_fisher_vp = empirical_fisher_vp_func({"loc": v})["loc"]
+
+    prec_matrix = torch.linalg.inv(cov_mat)
+    true_vp = prec_matrix.mv(v)
+
+    assert torch.all(torch.isclose(empirical_fisher_vp, true_vp, atol=0.1))
+
+
+@pytest.mark.parametrize(
+    "data_config",
+    [
+        (torch.zeros(1, requires_grad=True), torch.eye(1)),
+        (torch.ones(2, requires_grad=True), torch.eye(2)),
+    ],
+)
+def test_fisher_vmap_smoke(data_config):
+    loc, cov_mat = data_config
+    func_log_prob = gaussian_log_prob
+    log_prob_params = {"loc": loc}
+    N_monte_carlo = 10000
+    data = Predictive(GaussianModel(cov_mat), num_samples=N_monte_carlo)(loc)
+    empirical_fisher_vp_func = make_empirical_fisher_vp(
+        func_log_prob, log_prob_params, data, cov_mat=cov_mat
+    )
+    v_single = torch.ones(cov_mat.shape[1])
+    v_batch = torch.stack([v_single, v_single], axis=0)
+    empirical_fisher_vp_func_batched = torch.func.vmap(empirical_fisher_vp_func)
+
+    # Check if fisher vector product works on a single vector and a batch of vectors
+    empirical_fisher_vp_func({"loc": v_single})
+    empirical_fisher_vp_func_batched({"loc": v_batch})
+    try:
+        empirical_fisher_vp_func({"loc": v_batch})
+        assert False, "Fisher vector product should not work on a batch of vectors"
+    except RuntimeError:
+        pass
+    try:
+        empirical_fisher_vp_func_batched({"loc": v_single})
+        assert False, "Batched Fisher vector product should not work on a single vector"
```

Comment:
Can you use `with pytest.raises(RuntimeError)` contexts to check these failures instead of manual `try`/`except`?

Reply:
Done.